src/client/websocket/WebSocketShard.js

```diff
@@ -702,31 +774,51 @@ class WebSocketShard extends EventEmitter {
    * @private
    */
   destroy({ closeCode = 1_000, reset = false, emit = true, log = true } = {}) {
+    // Making the variable false to check for zombie connections.
+    this.closeEmitted = false;
+
     if (log) {
       this.debug(`[DESTROY]
     Close Code    : ${closeCode}
     Reset         : ${reset}
     Emit DESTROYED: ${emit}`);
     }
 
+    this.debug(`[WS Destroy] Step 0: Remove all timers.`);
     // Step 0: Remove all timers
     this.setHeartbeatTimer(-1);
     this.setHelloTimeout(-1);
 
+    this.debug(
+      `[WS Destroy] Step 1: Attempting to close the WebSocket. | WS State: ${
+        CONNECTION_STATE[this.connection?.readyState ?? CONNECTION_STATE]
```

Comment:
This doesn't look right 🤔
```suggestion
        CONNECTION_STATE[this.connection?.readyState ?? WebSocket.CLOSED]
```